indico/MaKaC/conference.py

```diff
@@ -2972,11 +2911,14 @@ def recoverSession(self, session, check, isCancelled):
         self.addSession(session, check, session.getId())
         session.recover(isCancelled)
 
-    def getSessionById( self, sessionId ):
+    def getSessionById(self, sessionId):
         """Returns the session from the conference list corresponding to the
             unique session id specified
         """
-        return self.sessions.get(sessionId,None)
+        from indico.modules.events.sessions.models.sessions import Session as Session_
+        if not sessionId.isdigit():
+            return None
+        return Session_.find_first(event_id=self.as_event.id, friendly_id=int(sessionId), is_deleted=False)
```

Comment:
`db.m.Session` might be nicer here than having `Session` and `Session_` in the same module

or you can just add https://gist.github.com/ThiefMaster/c226de524f528e68284966cb94522591 to `Event` and then return `self.as_event.get_session(friendly_id=sessionId)`